app/ui/design-system/src/lib/Components/NavigationBar/MobileMenu.tsx

```diff
@@ -1,39 +1,54 @@
 import { useState } from "react"
 import { MenuContent } from "./MenuContent"
 import { MobileMenuBackButton } from "./MobileMenuBackButton"
-import { MobileMenuTabbed } from "./MobileMenuTabbed"
 import { MobileMenuTopLevel } from "./MobileMenuTopLevel"
-import { DropdownMenuItem, isDropdownMenuItem, MenuItem } from "./types"
+import { DropdownMenu, DropdownMenuItem, LinkMenuItem, MenuItem } from "./types"
 
 export type MobileMenuProps = {
   menuItems: MenuItem[]
 }
 
 export function MobileMenu({ menuItems }: MobileMenuProps) {
-  const [selectedIndex, setSelectedIndex] = useState(-1)
-  const selectedMenuItem = menuItems[selectedIndex] as
-    | DropdownMenuItem
+  const [selectedIndex, setSelectedIndex] = useState<[number, number]>([-1, -1])
```

Comment:
This is a little confusing to me. Why do we need two indexes and what is the difference between them? Can we make this more clear maybe?